cmd/syncthing/main.go

```diff
@@ -810,6 +807,17 @@ func standbyMonitor(app *syncthing.App) {
 	}
 }
 
+func shouldUpgrade(cfg config.Wrapper, runtimeOptions RuntimeOptions) bool {
+	if opts := cfg.Options(); opts.AutoUpgradeIntervalH < 0 {
```

Comment:
We could bail early here if `upgrade.DisabledByCompilation` and then not have to handle `ErrUpgradeUnsupported` in the auto upgrade case?